src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -256,11 +256,34 @@ public override bool HtmlEncode
         }
 
         /// <inheritdoc />
+        public override bool IsBindingAllowed
+        {
+            get
+            {
+                if (MetadataKind == ModelMetadataKind.Property)
```

Comment:
these checks look redundant and cause the property to have defaults both here and in the `BindingMetadata` class.  is this done to avoid filling in the `BindingMetadata` property until and unless it's absolutely required?